lang/java/reef-utils/src/test/java/org/apache/reef/util/MultiAsyncToSyncTest.java

```diff
@@ -0,0 +1,240 @@
+    this.blocker = new MultiAsyncToSync(timeoutPeriodSeconds, SECONDS);
+    this.executor = Executors.newFixedThreadPool(2);
+  }
+
+  /**
+   * The one API call in our interface.
+   * @param input
+   * @return The input parameter incremented by one or zero for a timeout.
+   */
+  public Integer apiCall(final Integer input) {
+    // Create a future to run the asynchronous processing.
+    final long identifier = idCounter.getAndIncrement();
+    final FutureTask<Integer> task =
+        new FutureTask(new AsynchronousIncrementer(input, identifier, incrementerSleepTimeMillis, blocker));
+    taskQueue.add(task);
+
+    LOG.log(Level.INFO, "Running the incrementer...");
+    // Start the task and block the caller until it completes.
+    executor.execute(task);
+    if (blocker.block(identifier)) {
+      LOG.log(Level.INFO, "Call timed out...");
+      // Timeout occurred before the asynchronous processing completed.
+      return 0;
+    }
+    Integer result = 0;
+    try {
+      LOG.log(Level.INFO, "Call getting task result...");
+      result = task.get();
+    } catch (Exception e) {
+      result = -1;
+    }
+    return result;
+  }
+
+  /**
+   * Insure all test tasks have completed.
+   */
+  public void complete() {
+    try {
+      for (FutureTask<Integer> task : taskQueue) {
+        task.get();
+      }
+    } catch (Exception e) {
+      LOG.log(Level.INFO, "Caught exception waiting for completion...");
+    }
+  }
+}
+
+/**
+ * Verify proper operation of the MultiAsyncToSync class.
+ */
+public final class MultiAsyncToSyncTest {
+  private static final Logger LOG = Logger.getLogger(MultiAsyncToSyncTest.class.getName());
+
+  /**
+   * Verify calculations successfully complete when no timeout occurs.
+   */
+  @Test
+  public void testNoTimeout() {
+    LOG.log(Level.INFO, "Starting...");
+
+    // Parameters that do not force a timeout.
+    final int incrementerSleepTimeSeconds = 2;
+    final long timeoutPeriodSeconds = 4;
+    final Integer input = 1;
+
+    SynchronousApi apiObject = new SynchronousApi(incrementerSleepTimeSeconds, timeoutPeriodSeconds);
+    final Integer result = apiObject.apiCall(input);
+    apiObject.complete();
+
+    Assert.assertTrue(result.equals(input + 1));
+  }
+
+  /**
+   * Verify an error is returned when a timeout occurs.
+   */
+  @Test
+  public void testTimeout() {
+    LOG.log(Level.INFO, "Starting...");
+
+    // Parameters that do not force a timeout.
+    final int incrementerSleepTimeSeconds = 4;
+    final long timeoutPeriodSeconds = 2;
+    final Integer input = 1;
+
+    SynchronousApi apiObject = new SynchronousApi(incrementerSleepTimeSeconds, timeoutPeriodSeconds);
+    final Integer result = apiObject.apiCall(input);
+    apiObject.complete();
+
+    Assert.assertTrue(result.equals(0));
+  }
+
+  /**
+   * Used to make multiple asynchronous calls to the synchronous API.
+   */
+  final class AsynchronousCaller implements Callable<Boolean> {
```

Comment:
I moved it out and made it generic since it was originally specific to the test.